build_runner/test/generate/build_integration_test.dart

```diff
@@ -507,5 +508,33 @@ main() async {
       expect(result.stdout,
           contains('Missing dev dependecy on package:build_test'));
     });
+
+    test('Missing build_web_compilers dependency warns the user', () async {
+      await d.dir('a', [
+        await pubspec('a', currentIsolateDependencies: [
+          'build',
+          'build_config',
+          'build_resolvers',
+          'build_runner',
+        ]),
+        d.dir('web', [
+          d.file('a.dart', 'void main() {}'),
+        ]),
+      ]).create();
+
+      await pubGet('a');
+      var result = await startPub('a', 'run', args: ['build_runner', 'serve']);
+      var error = 'Missing dev dependency on package:build_web_compilers';
```

Comment:
I think you can do an `addTearDown(() => result.kill())` which would make it get killed even if the test times out or w/e.

Reply:
Nice idea, done.